src/defrag.c

```diff
@@ -924,219 +1320,48 @@ void computeDefragCycles(void) {
-            {server.pubsub_channels, defragPubsubScanCallback,
-             &(defragPubSubCtx){server.pubsub_channels, getClientPubSubChannels}},
-            {server.pubsubshard_channels, defragPubsubScanCallback,
-             &(defragPubSubCtx){server.pubsubshard_channels, getClientPubSubShardChannels}},
-        };
-        do {
-            int num_stages = sizeof(defrag_stages) / sizeof(defrag_stages[0]);
-            serverAssert(defrag_stage < num_stages);
-            defragStage *current_stage = &defrag_stages[defrag_stage];
-
-            /* before scanning the next bucket, see if we have big keys left from the previous bucket to scan */
-            if (defragLaterStep(db, slot, endtime)) {
-                quit = 1; /* time is up, we didn't finish all the work */
-                break;    /* this will exit the function and we'll continue on the next cycle */
-            }
-
-            if (!defrag_later_item_in_progress) {
-                /* Continue defragmentation from the previous stage.
-                 * If slot is -1, it means this stage starts from the first non-empty slot. */
-                if (slot == -1) slot = kvstoreGetFirstNonEmptyDictIndex(current_stage->kvs);
-                defrag_cursor = kvstoreDictScanDefrag(current_stage->kvs, slot, defrag_cursor, current_stage->scanfn,
-                                                      &defragfns, &(defragCtx){current_stage->privdata, slot});
-            }
+    // Defrag gets paused while a child process is active.  So there's no point in starting a new
+    //  cycle or adjusting the CPU percentage for an existing cycle.
+    if (hasActiveChildProcess()) return;
 
-            if (!defrag_cursor) {
-                /* Move to the next slot only if regular and large item scanning has been completed. */
-                if (listLength(db->defrag_later) > 0) {
-                    defrag_later_item_in_progress = 1;
-                    continue;
-                }
-
-                /* Move to the next slot in the current stage. If we've reached the end, move to the next stage. */
-                if ((slot = kvstoreGetNextNonEmptyDictIndex(current_stage->kvs, slot)) == -1) defrag_stage++;
-                defrag_later_item_in_progress = 0;
-            }
+    computeDefragCpuPercent(); // updates server.active_defrag_running (historical)
 
-            /* Check if all defragmentation stages have been processed.
-             * If so, mark as finished and reset the stage counter to move on to next database. */
-            if (defrag_stage == num_stages) {
-                all_stages_finished = 1;
-                defrag_stage = 0;
-            }
+    // Historically, server.active_defrag_running is both a logical boolean (indicating that a cycle
+    //  is running), and also an integer representation of the CPU percentage that should be applied.
+    //  A non-zero value here means that a cycle should be started, if it's not already running.
```

Comment:
```suggestion
    /* server.active_defrag_running is both a logical boolean (indicating that a cycle
    /* is running), and also an integer representation of the CPU percentage that should be applied.
    /* A non-zero value here means that a cycle should be started, if it's not already running. */
```

Historically is an awkward word choice here since it implies it's no longer true. 

Second more wider point, the general preference is to use c style comments `/* comment */` for everything,  especially multi-line comments but including single line comments.

Reply:
I'll fix the variable name.

Re: comments, I'll address all the multi-line comments.  However we ARE using `//` in a number of places throughout the code.  Is this something like the "bool hate"?